dev/src/write-batch.ts

```diff
@@ -103,6 +104,33 @@ export class WriteResult {
   }
 }
 
+/**
+ * A BatchWriteResult wraps the write time and status returned by Firestore
+ * when making BatchWriteRequests.
+ *
+ * @private
+ */
+export class BatchWriteResult {
```

Comment:
I think this type could be replaced with a typed array: Array<[WriteResult, Status]>. It would remove the conversion logic you had to add.

What do you think?

Reply:
I initially considered that, but opted not to because we would be indexing into the array to access WriteResult and Status, which seemed less readable than being able to directly access properties on an object

Reply:
Sounds reasonable.